keps/sig-network/3070-reserved-service-ip-range/README.md

```diff
@@ -0,0 +1,444 @@
+  - [Monitoring Requirements](#monitoring-requirements)
+  - [Dependencies](#dependencies)
+  - [Scalability](#scalability)
+  - [Troubleshooting](#troubleshooting)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests for meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Kubernetes Services are an abstract way to expose an application running on a
+set of Pods. Services has a ClusterIP that is virtual and allows to load-balance
+traffic across the different Pods. This ClusterIP can be assigned:
+
+- dynamically, the cluster will pick one within the configured Service IP range.
+- statically, the user will set one IP within the configured Service IP range.
+
+Currently, there is no possibility, before creating a Service with a static Cluster IP,
+to know if the IP has been chosen already by any other Service using dynamic allocation.
+
+The Service IP range can be logically subdivided to avoid the risk of conflict between
+Services that use static and dynamic IP allocation.
+
+## Motivation
+
+There are situation that users need to rely on well-known predefined IP addresses.
+
+The best example is the Kubernetes DNS, the IP is hardcoded in the kubelet flag `--cluster-dns`.
```

Comment:
(Well, no, the _best_ example is the apiserver...)

Reply:
is it possible to race with the apiserver .1 address? 
The loop to create the `kubernetes.default` runs at the beginning, I honestly don't know if at that point clients are able to create services, but is worth adding it